### ISSUE TITLE
Installation issue: gtkplus@3 cannot be installed

### ISSUE THREAD

**Issue by moguzhankarakaya** [LLNL/spack#27565]
### Steps to reproduce the issue

```
$ spack install gtkplus@3.24.29
==> Error: gtkplus@3.24.29 is unsatisfiable, conflicts are:
  no version satisfies the given constraints
```
or
```
$ spack spec -N gtkplus@3.24.29
...
==> Error: gtkplus@3.24.29 is unsatisfiable, conflicts are:
  no version satisfies the given constraints
```

### Information on your system

Just git cloned the development branch and run the above code. 

 

### Additional information

The reason is as following:

* `gtkplus@3` depends on `at-spi2-atk` and it depends on `at-spi2-core`. Then `at-spi2-core` depends on `dbus`. Finally `dbus` depends on `docbook-xml@4.4`. It explicitly requires docbook-xml version `4.4`. This change was added 4 days ago with [PR#27420](https://github.com/spack/spack/pull/27420). 

* On the other hand, all versions of `gtkplus` depends of `librsvg` and it depends on `gtk-doc`. Unfortunately `gtk-doc` depends on `docbook-xml@4.3`. It specifically requires version `4.3`.  This dependency is introduced by [PR#22228](https://github.com/spack/spack/pull/22228) 8 months ago.

* As a result of those above mentioned two dependencies of `gtkplus@3` it cannot be installed due to conflicting `docbook-xml` versions.

My suggestion is if it is possible one of the packages should give a range of versions for `docbook-xml` or free it up completely. Sadly non of those `gtkplus`, `gtk-doc` or `dbus` has a maintainer showing up when I did `spack maintainers ...` on them.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by moguzhankarakaya** [comment]
I am sorry since my issue doesn't follow the guidelines really well. However, I debugged the problem and noted down two key PRs that caused the issue in the first place. If you agree my suggestion above, I would gladly open a PR fixing this issue myself.

**Comment by alalazo** [comment]
@moguzhankarakaya Feel free to submit a PR to fix this and ping me. I marked the issue as "concretizer-sue-case" since this would be solved if we had separate concretization of build dependencies (in which case one node would use `dockbook-xml@4.3` and the other `dockbook-xml@4.4`)

**Comment by wdconinc** [comment]
Both `librsvg` and `libcroco` (which also build-depends on `gtk-doc`) have `--enable-gtk-doc=no` configure options. I think that may present a pre-v0.18 solution to this issue.

**Comment by wdconinc** [comment]
@moguzhankarakaya Feedback on #27790 welcome.

**Comment by alalazo** [comment]
I confirm this specific issue was solved in #27790
![Screenshot from 2022-07-04 15-55-27](https://user-images.githubusercontent.com/4199709/177169314-b3dce0ca-582f-4e02-9e37-bc46a15c720c.png)
